src/discord/discord.service.ts

```diff
@@ -1,75 +1,91 @@
-import { Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { HttpException, HttpStatus } from '@nestjs/common';
+import { createClient } from '@astrajs/collections';
+
 import { ReadDiscordDto } from './dto/read-discord.dto';
 import { UpdateDiscordDto } from './dto/update-discord.dto';
 
-@Injectable()
 export class DiscordService {
-  private discord: ReadDiscordDto[] = [];
-  create(createDiscordDto: ReadDiscordDto) {
+  astraClient;
+  usersCollection;
+  basePath = '/api/rest/v2/KEYSPACES/<namespace>/collections/<collectionName>';
+  constructor() {
+    (async () => {
+      try {
+        this.astraClient = await createClient({
+          astraDatabaseId: process.env.ASTRA_DB_ID,
+          astraDatabaseRegion: process.env.ASTRA_DB_REGION,
+          applicationToken: process.env.ASTRA_DB_APPLICATION_TOKEN,
+        });
+
+        this.usersCollection = this.astraClient
+          .namespace('crud')
+          .collection('users');
+      } catch (ex) {
+        console.log(ex.message);
+      }
+    })();
+  }
+  async create(createDiscordDto: ReadDiscordDto) {
     const discordUser = {
-      id: Date.now(),
       username: createDiscordDto.username,
       bio: createDiscordDto.bio,
       socials: { ...createDiscordDto.socials },
       createdOn: new Date(Date.now()),
       updatedOn: new Date(Date.now()),
     };
+
     if (!discordUser.username) {
       throw new HttpException('Incomplete Data', HttpStatus.BAD_REQUEST);
     }
-    this.discord.push(discordUser);
-    return 'User added successfully!';
+    await this.usersCollection.create({
+      ...discordUser,
+    });
+    return 'User added successfully! ';
   }
 
-  findAll() {
-    return [...this.discord];
+  async findAll() {
+    return await this.usersCollection.find();
```

Comment:
I think NestJS resolves the promise in the `controller` and from the service a pro use or observable should be returned - maybe we be consistent and use observables